aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/config/DaemonConfiguration.java

```diff
@@ -101,6 +101,11 @@ private boolean setUDPAndTCPAddress(@Nullable String addr, boolean ignoreInvalid
         boolean result = false;
         try {
             result = processAddress(addr);
+
+            if (!result) {
```

Comment:
Don't we need to check `ignoreInvalid`?

Reply:
I don't think so, because the exception will always be caught and then `ignoreInvalid` is checked in the catch block. If you think there's a more idiomatic way of doing this I'd be open to it though.

Reply:
Ah yeah didn't notice it - it's a bit weird to throw and immediately catch an exception. Outside of the try block having this with `if (!result && !ignoreInvalid)` throw might be more readable, but it's marginable at best.

Reply:
I actually just did a bigger refactor and made the `processAddress` method `void`, and just had it throw instead of return false. IMO this improved readability in both the `setUDPandTCPAddress` and `processAddress` methods, let me know what you think.